src/main/scala/circt/stage/phases/CIRCT.scala

```diff
@@ -136,39 +122,65 @@ class CIRCT extends Phase {
         /* Communicate the annotation file through a file. */
         (outputAnnotationFileName.map(a => Seq("-annotation-file", a))).getOrElse(Seq.empty) ++
         /* Convert the target to a firtool-compatible option. */
-        (circtOptions.target match {
-           case Some(CIRCTTarget.FIRRTL) => Seq("-ir-fir")
-           case Some(CIRCTTarget.HW) => Seq("-ir-hw")
-           case Some(CIRCTTarget.Verilog) => None
-           case Some(CIRCTTarget.SystemVerilog) => None
-           case None => throw new Exception(
-             "No 'circtOptions.target' specified. This should be impossible if dependencies are satisfied!"
-           )
-         })
+        ((circtOptions.target, split) match {
+          case (Some(CIRCTTarget.FIRRTL), false)        => Seq("-ir-fir")
+          case (Some(CIRCTTarget.HW), false)            => Seq("-ir-hw")
+          case (Some(CIRCTTarget.Verilog), true)        => Seq("--split-verilog")
+          case (Some(CIRCTTarget.Verilog), false)       => None
+          case (Some(CIRCTTarget.SystemVerilog), true)  => Seq("--split-verilog")
+          case (Some(CIRCTTarget.SystemVerilog), false) => None
+          case (None, _)                                =>
+            throw new Exception(
+              "No 'circtOptions.target' specified. This should be impossible if dependencies are satisfied!"
+            )
+          case _                                        =>
+            throw new Exception(
+              s"Invalid combination of circtOptions.target ${circtOptions.target} and split ${split}"
+            )
```

Comment:
I would change this to:

```scala
case (_, false) =>
```

This makes it explicit what the check is doing as opposed to it being the fallback case.

The error message would be better as: "The `circtOptions.target` specified (${circtOptions.target}) does not support running with an `EmitAllModulesAnnotation` as CIRCT only supports one-file-per-module for Verilog targets."

Reply:
shouldnt that error message then be case (_, true) => ?

Reply:
Yep. 😅